provisioner/provisioner.go

```diff
@@ -444,107 +411,171 @@ func (p *Provisioner) executeMondoo(ctx context.Context, ui packer.Ui, comm pack
 	for k := range p.config.Labels {
 		p.config.Annotations[k] = p.config.Labels[k]
 	}
+
 	// build configuration
-	connection := fmt.Sprintf("%s://%s", connType, endpoint)
-	if user != "" {
-		connection = fmt.Sprintf("%s://%s@%s", connType, user, endpoint)
-	}
-
-	conf := &VulnOpts{
-		Assets: []*Asset{
-			{
-				Name:         p.config.AssetName,
-				Connection:   connection,
-				IdentityFile: privKeyFile,
-				Password:     password,
-				Annotations:  p.config.Annotations,
-			},
+	conf := inventory.New(inventory.WithAssets(&asset.Asset{
+		Name:        p.config.AssetName,
+		Connections: []*providers.Config{assetConfig},
+		Annotations: p.config.Annotations,
+		Labels: map[string]string{
+			"packer.io/buildname": p.config.PackerBuildName,
+			"packer.io/buildtype": p.config.PackerBuilderType,
 		},
-		Insecure: true, // we do not check the hostkey for the packer build
+	}))
+
+	scanJob := &scan.Job{
+		Inventory:     conf,
+		Bundle:        policyBundle,
+		PolicyFilters: policyFilters,
+		ReportType:    scan.ReportType_FULL,
 	}
 
-	if p.config.Sudo != nil && p.config.Sudo.Active {
-		ui.Say("activated sudo")
-		conf.Sudo = VulnOptsSudo{
-			Active: p.config.Sudo.Active,
-		}
+	debugLogBuffer := &bytes.Buffer{}
+	logger.SetWriter(debugLogBuffer)
+	if p.config.Debug {
+		data, _ := json.Marshal(scanJob)
+		ui.Message(fmt.Sprintf("mondoo configuration: %v", string(data)))
+
+		// configure stderr logger
+		logger.Set("debug")
 	}
 
-	// pass incognito to mondoo scan
-	conf.Incognito = p.config.Incognito
+	// base 64 config env setting has always precedence
+	viper.SetConfigType("yaml")
+	if value := os.Getenv("MONDOO_CONFIG_BASE64"); len(value) > 0 {
```

Comment:
Going back to my first comment about `MondooConfigPath` - this looks like a place where we might want to use it?

Reply:
I am still not understanding what this means